### ISSUE TITLE
nnUNet commands not found in terminal

### ISSUE THREAD

**Issue by Hball99** [MIC-DKFZ/nnUNet#541]
Hello there!

Yesterday I installed nnUNet on my Linux based system according to your How-To-Installation guide. First I installed pytorch, then nnunet via "pip3 install nnunet". Everything seemed to work fine, I defined the environment variables, then I wanted to start with the Hippocampus example. 

My problem is that, when I call the "nnUNet_convert_decathlon_task -h" command, the terminal gives me the following: 
"zsh: command not found: nnUNet_convert_decathlon_task".
I tried calling "python3 -c "import nnunet"" from my $HOME directory, which is working totally fine. I get 

"Please cite the following paper when using nnUNet:
Isensee, F., Jaeger, P.F., Kohl, S.A.A. et al. "nnU-Net: a self-configuring method for deep learning-based biomedical image segmentation." Nat Methods (2020). https://doi.org/10.1038/s41592-020-01008-z
If you have questions or suggestions, feel free to open an issue at https://github.com/MIC-DKFZ/nnUNet"

as a response. So I assume that nnunet is installed correctly, at least to some extend. I also tried to install nnunet via the cloned github repository and in or outside of virtual environments. Everytime, the installation works perfectly fine, but when calling the commands, they are not found.

To double-check, I installed nnunet on my private windows-laptop, using a Ubuntu-20.4 sub-system. Everything worked totally fine, the commands worked instantly. 

So I have no idea, why those commands cannot be found on my linux system, when the installation apparently worked fine. Maybe there are missing PATH variables? 

I would appreciate it very much, if you could help me with my problem. Thanks!

**Comment by FabianIsensee** [comment]
Mhm interesting. Have you tried good-old bash instead of zsh? That might solve it.
Best,
Fabian

**Comment by Hball99** [comment]
Thanks for the quick reply!

I changed to bash, but the porblem ramains. Only difference is that I get the answer in german.

bash-4.2$ nnUNet_convert_decathlon_task -h
zsh: nnUNet_convert_decathlon_task: Befehl nicht gefunden...

I am wondering, why there is still an "zsh:" in front, even though "bash-4.2" shell is used

**Comment by FabianIsensee** [comment]
haha not sure why it should change languages. nnU-Net is all english.
It seems to be related to your zsh somehow, especially because it still seems to be looking for the command using zsh. Maybe create a completely new virtualenv with bash?

**Comment by Hball99** [comment]
I have created a new environment with bash. Now its actually looking for the command using bash, but cannot find it whatsoever:

bash-4.2$ nnUNet_convert_decathlon_task -h
bash: nnUNet_convert_decathlon_task: Kommando nicht gefunden.

Any other idea, what could be the reason for that behaviour? Otherwise I probably need to check with the system administrators...

**Comment by FabianIsensee** [comment]
unfortunately no but you can also use all the nnU-Net commands directly with python: `python nnunet/run/run_training.py` is the same as nnUNet_train. Have a look at the setup.py. That is where the nnUNet_* entry points are defined

**Comment by Hball99** [comment]
Thank you! That did the trick, and it seems that everything is working for now!

**Comment by FabianIsensee** [comment]
great!

**Comment by Hball99** [comment]
Hi Fabian!

You mentioned that python nnunet/run/run_training.py is the same as nnUNet_train. That works perfectly fine.
But my question is now, how do I call a function, that is not the main(). For example the following: 

nnUNet_download_pretrained_model Task005_Prostate

That corresponds to calling:

nnunet.inference.pretrained_models.download_pretrained_model:print_pretrained_model_requirements

But how could I call a specific function inside the python script? I tried to go into the folder "pretrained models", then called the python shell via "pyhton3" and then I called "from download_pretrained_model import download_by_name". Up to that point, everything worked fine. But now I dont know how to call the function, in order to download Task005_Prostate. I tried several options, like "download_by_name() Task005_Prostate" or  "download_by_name(Task005_Prostate)" but none worked. When just calling download_by_name(), I got the message 

"usage: [-h] task_name
: error: the following arguments are required: task_name"

Do you know, how to properly call the function inside the script?
Thanks!

**Comment by Hball99** [comment]
I think I found a way myself! So never mind my question for now :-)

**Comment by FabianIsensee** [comment]
Great :-)
Best,
Fabian

**Comment by natacanta5** [comment]
> I think I found a way myself! So never mind my question for now :-)

Hi¡ I have the same problem, in bash it does not find the commands nnUNet_train, could you explain me how to solve it please?

**Comment by FabianIsensee** [comment]
tagging @Hball99 :-)

**Comment by Hball99** [comment]
In the end my problem boiled down to my virtual environment being defined faulty by me. I did set up a new virtual environment using virtualenv and everything worked. I think I used the command "venv my_env" (if I remember correctly), eventhough I should have used " python3 -m venv my_env". Since the environment seemed to work perfectly fined, I would have never assumed that that was my problem.. Took me quite some time haha :D
